vmdb/app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -477,6 +474,17 @@ def build_template_grid(templates,sort_order="DESC",sort_by="name")
     @temp[:templates] = sorted.uniq
   end
 
+  def dialog_partial_for_workflow
+    case (@edit || @options).try(:[], :wf)
+    when MiqProvisionWorkflow
+      "prov_dialog"
+    when MiqHostProvisionWorkflow
```

Comment:
Do you want a `when` here, or does this belong in an `else`?

The original code looks like the `"prov_host_dialog"` partial is always rendered.

Not sure if this is fixing a bug or a simple refactor.

+1 on fixing the `kind_of?`